wave_geometry/include/wave/geometry/exception_helpers.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef WAVE_EXCEPTION_HELPERS_HPP
+#define WAVE_EXCEPTION_HELPERS_HPP
+
+#include "wave/geometry/rotation.hpp"
+#include "wave/utils/math.hpp"
+
+namespace wave {
+
+
+// This function checks to make sure all elements of the input
+// Eigen array have finite values (not Nan or +/- Inf), as these
+// values are invalid for rotation geometry.
+template <typename MatrixType>
+void checkArrayFinite(const Eigen::MatrixBase<MatrixType> &mat) {
```

Comment:
I think we already differentiate class methods using the `this->` keyword. Functions can have `camelCase`.